src/main.cpp

```diff
@@ -783,6 +793,31 @@ int main ( int argc, char** argv )
                 strServerBindIP = "";
             }
         }
+
+#ifndef NO_JSON_RPC
+        //
+        // strJsonRpcBind address defaults to loopback and should not be empty, but
+        // in the odd chance that an empty IP is passed, we'll check for it here.
+
+        if ( strJsonRpcBindIP.trimmed().isEmpty() )
+        {
+            qCritical() << qUtf8Printable ( QString ( "JSON-RPC is enabled but no bind address was provided, exiting." ) );
+            exit ( 1 );
+        }
+        else
+        {
+            // This means of testing the validity of IP addresses is far from perfect but
+            // we do it here as an upfront check.  The downstream network calls will error
+            // out on malformed addresses not caught here.
+
+            QHostAddress InetAddr;
+            if ( !InetAddr.setAddress ( strJsonRpcBindIP ) )
```

Comment:
We should introduce a method which checks IP validity e.g via Regex

Reply:
You're probably right -- an IP address validation function used through the codebase.  I think that is out of scope for this particular PR, however.

Reply:
I think using the same methods as for Server Bind IP will be okay for now.